File: app/src/main/java/fr/gstraymond/android/CardDetailActivity.java
package fr.gstraymond.android;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.TextView;

import fr.gstraymond.R;
import fr.gstraymond.android.fragment.CardDetailFragment;
import fr.gstraymond.models.response.Card;
import fr.gstraymond.tools.LanguageUtil;

import static fr.gstraymond.constants.Consts.CARD;
import static fr.gstraymond.constants.Consts.POSITION;

public class CardDetailActivity extends CardCommonActivity implements
        CardDetailFragment.Callbacks {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_card_detail);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        TextView title = (TextView) findViewById(R.id.toolbar_title);
        title.setText(formatTitle(this, getCard()));

        replaceFragment(new CardDetailFragment(), R.id.card_detail_container, getBundle());
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {

            case R.id.pictures_tab:
                Intent intent = new Intent(this, CardPagerActivity.class);
                intent.putExtra(CARD, getCard());
                startActivity(intent);
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.card_detail_menu, menu);
        return true;
    }

    @Override
    public void onItemSelected(int id) {
        Intent intent = new Intent(this, CardPagerActivity.class);
        intent.putExtra(CARD, getCard());
        // first element is a card
        intent.putExtra(POSITION, id - 1);
        startActivity(intent);
    }

    public static String formatTitle(Context context, Card card) {
        if (LanguageUtil.showFrench(context) && card.getFrenchTitle() != null) {
            return String.format("%s (%s)", card.getFrenchTitle(), card.getTitle());
        }

        return card.getTitle();
    }
}

File: app/src/main/kotlin/fr/gstraymond/android/CardDetailActivity.kt
package fr.gstraymond.android

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.TextView
import fr.gstraymond.R
import fr.gstraymond.affiliate.ebay.LinkGenerator
import fr.gstraymond.android.fragment.CardDetailFragment
import fr.gstraymond.models.search.response.Card
import fr.gstraymond.models.search.response.getLocalizedTitle
import fr.gstraymond.ui.adapter.CardDetailViews
import fr.gstraymond.ui.view.impl.ShareView
import fr.gstraymond.utils.app
import fr.gstraymond.utils.find
import fr.gstraymond.utils.startActivity

class CardDetailActivity : CardCommonActivity(R.layout.activity_card_detail),
        CardDetailFragment.Callbacks {

    companion object {
        fun getIntent(context: Context, card: Card) =
                Intent(context, CardDetailActivity::class.java).apply {
                    putExtra(CARD_EXTRA, card)
                }
    }

    private val rootView by lazy { find<View>(android.R.id.content) }
    private val picsView by lazy { find<TextView>(R.id.card_detail_pics) }
    private val ebayView by lazy { find<TextView>(R.id.card_detail_ebay) }
    private val deckView by lazy { find<TextView>(R.id.card_detail_deck) }
    private val favoriteView by lazy { CardDetailViews(app(), this, ShareViewCallbacks()) }

    private inner class ShareViewCallbacks : ShareView.ShareViewCallbacks {

        override fun wishlistChanged(position: Int) {}
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setSupportActionBar(find(R.id.toolbar))
        true.actionBarSetDisplayHomeAsUpEnabled()

        val title = find<TextView>(R.id.toolbar_title)
        title.text = card.getLocalizedTitle(this, Card::title) { c, ft -> "$ft (${c.title})" }

        replaceFragment(CardDetailFragment(), R.id.card_detail_container, getBundle())

        picsView.setOnClickListener {
            startActivity { CardPagerActivity.getIntent(this, card) }
        }
        
        ebayView.setOnClickListener {
            startActivity {
                Intent(Intent.ACTION_VIEW).apply {
                    data = Uri.parse(LinkGenerator.generate(card.title))
                }
            }
        }

        deckView.setOnClickListener {
            startActivity { CardDeckActivity.getIntent(this, card) }
        }
    }

    override fun onResume() {
        super.onResume()
        favoriteView.display(rootView, card, 0)
    }

    override fun onItemSelected(id: Int) = startActivity {
        CardPagerActivity.getIntent(this, card, id)
    }

    override fun onListSelected(list: String) {
        startActivity {
            when (list) {
                "wishlist" -> WishListActivity.getIntent(this)
                else -> DeckDetailActivity.getIntent(this, list)
            }
        }
    }
}
